master/internal/sproto/agent_resource_manager.go

```diff
@@ -112,3 +112,37 @@ func (t TerminateDecision) String() string {
 	}
 	return strings.Join(item, ",")
 }
+
+// ValidateResourcePoolAvailabilityParam contains the params for ValidateResourcePoolAvailability().
+type ValidateResourcePoolAvailabilityParam struct {
+	Name   string
+	Slots  int
+	TaskID *model.TaskID
+}
+
+// ValidateResourcePoolAvailabilityParamOption accepts functions that may modify the
+// *ValidateResourcePoolAvailabilityParam instance.
+type ValidateResourcePoolAvailabilityParamOption func(v *ValidateResourcePoolAvailabilityParam)
```

Comment:
non blocking: These helpers seem like a little overkill to me. I'm not against them but I think just the struct would be fine for me too.